### DISCUSSION TITLE
How to open a link into a new tab? (i.e. do target="_blank")

### DISCUSSION THREAD

**Discussion by lyderic** [sqlpage/SQLPage#379] [Q&A]
I have some links in a table that'd need to be opened in a new tab when clicked.

This is what I have tried, it doesn't work and I have no other idea:

```sql
SELECT 'table' AS component, 'URL' AS markdown;
SELECT
  name AS Name,
  FORMAT('<a href="%s" target="_blank">%s</a>', url, url) AS URL
FROM bookmarks;
```

The embedded HTML is displayed as is:

![Screenshot from 2024-06-06 15-15-59](https://github.com/lovasoa/SQLpage/assets/5059957/095a0282-e0ff-41b4-af63-41bbae2ecf01)

This works as expected, but it doesn't open links in a new tab:

```sql
SELECT 'table' AS component, 'URL' AS markdown;
SELECT
  name AS Name,
  FORMAT('[%s](%s)', name, url) AS URL
FROM bookmarks;
```

SQLpage uses [markdown-rs](https://github.com/wooorm/markdown-rs) to process markdown. This dependency is supposed to be CommonMark compliant and AFAIK, CommonMark supports HTML tags.

I know that I can roll out my own handlebar template, but I'd like to have your ideas/suggestions/hacks to avoid doing that.

L.

**Comment by lovasoa** [comment]
SQLPage uses [github flavored markdown](https://github.github.com/gfm/).
I am a bit reluctant to offer raw html, for security reasons...

This answer may be a little disappointing, but: you can always open normal links in a new tab with a center mouse click if you want to. You cannot open open `target=_blank` links in the same tab when you want to. So, [it's often better to avoid it](https://css-tricks.com/use-target_blank/)

**Comment by lyderic** [comment]
Thanks for coming back to me.

I read the linked article with interest. There are points I hadn't considered, some I agree with, some I don't.

That said, I appreciate your reluctance to offer raw html and I support it.

Otherwise, please note that github's markdown _actually_ supports html tags. For example, `<i>italic</i> and <b>bold</b>` or `<a href="foo.com">foo</a>` work in a github's README.md. However, github also [disallows target attributes](https://stackoverflow.com/questions/41915571/open-link-in-new-tab-with-github-markdown-using-target-blank), so `<a href="foo.com" target="_blank">foo</a>` _doesn't_ work, which kind of shows that they are frowned upon.

For the moment, I will live without `target="_blank"` in my links, this is not a big problem.

**Background**: I am building a bookmarks application. Opening links in new tabs looks like a legitimate use case for this context. However, I agree that middle-clicking, Ctrl-clicking, or using similar methods is easy and intuitive enough.

Cheers,
L.

**Reply by lovasoa** [reply]
I hope you'll present us your bookmark app when it's ready !

**Comment by lyderic** [comment]
Yes I will, but please keep in mind this is a very modest personal project.
Not even multiuser ;-)

**Reply by lovasoa** [reply]
I think it's nice for all users to get inspiration from each other, and to keep an eye on what others are doing, from personal projects, to enterprise apps, to blogs and websites.

**Comment by lovasoa** [comment]
Hi @lyderic . The markdown interpreter still doesn't allow new tab links, but I added a `target` row-level property to the button component to open links in new tabs: https://sql.ophir.dev/documentation.sql?component=button#component

**Comment by lyderic** [comment]
Excellent. Many thanks 😊
